### DISCUSSION TITLE
Questions about the Intel Arc support

### DISCUSSION THREAD

**Discussion by cyatarow** [vladmandic/sdnext#998] [Q&A]
*(edit: 2023-05-16 12:26 UTC)*

- Can this repo work on ~~WSL~~*Windows (including WSL)* with Arc?
- If yes, how fast and how large an image can it generate?
- Are there ~~WSL~~*Windows(or WSL)*-specific bugs (e.g. memory leak) that are not present in native Linux environments?

If I don't have to use native Linux, I'd love to.

**Comment by vladmandic** [comment]
per intel's support article <https://www.intel.com/content/www/us/en/support/articles/000093216/graphics.html>, arc does not support gpu virtualization required to access it inside wsl2.
how fast is it? seems pretty fast for its price range, A770 performs roughly on the level of RTX3070

**Reply by cyatarow** [reply]
Thank you, but sorry, I changed the questions a little.
And it would be nice to know about the maximum resolution that can be generated.

**Reply by vladmandic** [reply]
i honestly don't know, sorry.

**Reply by cyatarow** [reply]
To be honest, there is very little information on running SD with Arc, but I'll keep gathering it too.

**Comment by Tillerz** [comment]
Isn't the base resolution always 512x512 (and up to 768x768 for SD 2.x)? All else is upscaled, and if you need a really large image, (some of) those upscalers can do tiled upscaling, so the whole image is broken down into smaller segments for rendering.

If I do a direct upscale when rendering, I can do 2040x028 with SPD disabled, that takes all the memory of my card. so I'd do tiled upscaling, too.


Some numbers from my PC (nVidia RTX 4090, 24GB VRAM) just to show how the memory consumption and time behaves if you directly upscale while rendering the image:

rendering 512x512 (no upscaling):
`Time taken: 3.86s | GPU active 2823 MB reserved 3138 MB | System peak 4822 MB total 24564 MB`

rendering 1280x1280 during rendering (hires fix, upscaler: Latent):
`Time taken: 9.30s | GPU active 5686 MB reserved 14180 MB | System peak 15864 MB total 24564 MB`

rendering 2048x2048 during rendering (hires fix, upscaler: Latent):
`Time taken: 38.39s | GPU active 19049 MB reserved 23470 MB | System peak 24564 MB total 24564 MB`

upscaling to 4096x4096 in Process Image (from 512x512, upscaler: SwinIR_4x):
`Time taken: 1m 11.39s | GPU active 14035 MB reserved 14064 MB | System peak 15752 MB total 24564 MB`


You can see when doing tiled upscaling it uses less memory than for a direct upscale to 2048x2048.

**Comment by Disty0** [comment]
WSL2 does work but it has memory leak issues.
I don't use Windows so i never used WSL, I don't have any issues with Linux.

You can generate 2048x2048 with sub-quad.

512x512 (20 samples):
`Time taken: 5.69s | GPU active 4779 MB reserved 5064 MB | System peak 4437 MB total 15474 MB`

512x512 to 1280x1280 (20 samples) with img2img:
`Time taken: 29.12s | GPU active 7461 MB reserved 8780 MB | System peak 5073 MB total 15474 MB`

512x512 to 2048x2048 (20 samples) with img2img:
It is unstable at 2048x2048.
`Time taken: 4m 0.93s | GPU active 12456 MB reserved 15920 MB | System peak 7389 MB total 15474 MB`

**Reply by cyatarow** [reply]
[According to this PR](https://github.com/vladmandic/automatic/pull/643), you are using A770 16GB.
How about A750? I'm trying to decide whether to buy A770 or A750.

**Reply by Disty0** [reply]
They are similar in performance but do not buy an 8 GB GPU in 2023.
